code/modules/vehicles/mecha/combat/justice.dm

```diff
@@ -207,13 +646,12 @@
 
 ///Called when invisibility activated.
 /datum/action/vehicle/sealed/mecha/invisibility/proc/invisibility_on()
+	if(SEND_SIGNAL(chassis, COMSIG_JUSTICE_INVISIBILITY_ACTIVATE, src) & COMPONENT_CANCEL_JUSTICE_INVISIBILITY_ACTIVATE)
```

Comment:
These are some clean signals, but they should be simple proc calls if you're doing it all on the same type